_drafts/practical_zephyr_01_kconfig.md

```diff
@@ -0,0 +1,986 @@
+
+The reason is that `Kconfig` files can have _dependencies_: Enabling one option can automatically also enable other options. Sadly, at the time of writing, _Kconfig_ does not seem to warn about conflicting values for symbols, e.g., from the application configuration file: It'd be great if Kconfig would tell me that my `CONFIG_PRINTK=n` setting is overwritten by some other option and has no effect!
+
+Zephyr contains _hundreds_ of so-called Kconfig _fragments_. It is therefore almost impossible to navigate these files without tool support. Thankfully, _West_ integrates two graphical tools to explore available _Kconfig_ options and their current setting: `menuconfig` and `guiconfig`.
+
+
+### Loading `menuconfig` and `guiconfig`
+
+For the `build` command, _West_ has some builtin _targets_, two of which are used for _Kconfig_:
+
+```bash
+$ west build --build-dir ../build -t usage
+-- west build: running target usage
+...
+Kconfig targets:
+  menuconfig - Update .config using a console-based interface
+  guiconfig  - Update .config using a graphical interface
+...
+```
+
+The targets `menuconfig` and `guiconfig` can be used to start graphical _Kconfig_ tools which allow modifying the `.config` file in your build directory. Both tools are essentially identical. The only difference is that `menuconfig` is a text-based graphical interface that opens directly in your terminal, whereas `guiconfig` is a "clickable" user interface running in its own window. The screenshots below show the tools launched by the following commands:
+
+```bash
+$ west build --build-dir ../build -t menuconfig
+$ west build --build-dir ../build -t guiconfig
+```
+
+> **Notice:** If you did not build your project or if the project does not compile, then the tool will not load since the `zephyr/.config` file is missing.
+
+![]({% img_url practical-zephyr/kconfig-menuconfig.png %})
+![]({% img_url practical-zephyr/kconfig-guiconfig.png %})
+
+For the sake of simplicity, we'll use `menuconfig` in the remainder of this article. In both tools, you can search for configuration symbols. In `menuconfig` - similar to `vim` - you can use the forward slash `/` to start searching, e.g., for our `PRINTK` symbol. Several results match, but only one of them is the plain symbol `PRINTK` which also shows the same description that we've seen in the `Kconfig` file:
+
+```
+PRINTK(=y) "Send printk() to console"
+```
+
+Navigate to the symbol either using the search or using the path _Subsystems and OS Services > Debugging Options_. You should see the following screen:
+
+![]({% img_url practical-zephyr/kconfig-menuconfig-debug.png %})
+
+Typically, you can toggle `bool` symbols using `Space` or `Enter`, but this does not seem to be possible for `PRINTK`. You can see that a symbol cannot be changed - which is indicated by the marker `*` being enclosed by dashes `-*-` instead of square brackets `[*]`. Using `?` we can display the symbol's information:
```

Comment:
One thing I hit often is after searching and selecting an option, the hidden Kconfigs will be shown in `menuconfig`. I think mentioning these can be toggled with `Shift+A` somewhere in this section would be great

Reply:
Good one!